src/vmm/src/persist.rs

```diff
@@ -249,7 +249,13 @@ fn snapshot_memory_to_file(
                 .dump_dirty(&mut file, &dirty_bitmap)
                 .map_err(Memory)
         }
-        SnapshotType::Full => vmm.guest_memory().dump(&mut file).map_err(Memory),
+        SnapshotType::Full => {
+            let dirty_bitmap = vmm.get_dirty_bitmap().map_err(DirtyBitmap)?;
+            vmm.guest_memory()
+                .dump_dirty(&mut file, &dirty_bitmap)
+                .map_err(Memory)
+        }
```

Comment:
This replicates the behavior of diff snapshots, it won't output the non-dirty pages.

It should output all pages when taking a full snapshot e.g.
```suggestion
        SnapshotType::Full => {
            let _ = vmm.get_dirty_bitmap().map_err(DirtyBitmap)?;
            vmm.guest_memory()
                .dump(&mut file)
                .map_err(Memory)
        }
```